File: Obj-C/Lab7-ListDemo/Lab7-ListDemo/ViewController.m
//
//  ViewController.m
//  Lab7-ListDemo
//
//  Created by 張力元 on 2019/3/4.
//  Copyright © 2019 張力元. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()
@property (strong, nonatomic) IBOutlet UIBarButtonItem *swichBtn;
@property (strong, nonatomic) IBOutlet UIView *tableViewContainer;
@property (strong, nonatomic) IBOutlet UIView *collectionViewContainer;
@end

@implementation ViewController
- (void)viewDidLoad {
    [super viewDidLoad];
    //一開始將Collection View的功能隱藏
    _collectionViewContainer.hidden = true;
    //初始化水果名稱陣列
    NSArray *friutArr = [[NSArray alloc]initWithObjects:@"apple",@"avocado",@"banana",@"cherries",@"coconut",@"grapes",@"lemon",@"orange",@"peach",@"pineapple",@"strawberry",@"tomato", nil];
    //儲存進UserDefault這個Dictionary中，方便之後讀取資料
    NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
    [userDefaults setObject:friutArr forKey:@"friutName"];
}

- (IBAction)swichBtnClick:(id)sender {
    //判斷現在顯示的是TableView 還是CollectionView
    if (_swichBtn.tag == 0) {
        //顯示Collection View
        _swichBtn.tag = 1;
        _collectionViewContainer.hidden = false;
        _tableViewContainer.hidden = true;
        self.title = @"Demo Collection View";
    }
    else{
        //顯示Table View
        _swichBtn.tag = 0;
        _collectionViewContainer.hidden = true;
        _tableViewContainer.hidden = false;
        self.title = @"Demo Table View";
    }
}
@end

File: Swift/lab7/lab7_exercise/lab7_exercise/ViewController.swift
//
//  ViewController.swift
//  lab7_exercise
//
//  Created by taizhou on 2019/7/30.
//  Copyright © 2019 taizhou. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    @IBOutlet var tableViewContainer: UIView!
    @IBOutlet var collectionViewContainer: UIView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        //一開始將Collection View的功能隱藏
        collectionViewContainer.isHidden = true
        //初始化水果名稱陣列
        let fruitNameArr = ["apple", "avocado", "banana", "cherries", "coconut", "grapes", "lemon", "orange", "peach", "pineapple", "strawberry", "tomato"]
        let userDefault = UserDefaults.standard
        userDefault.set(fruitNameArr, forKey: "fruitName")
        //設置此頁面的標題
        title = "demo table view"
    }
    
    @IBAction func switchBtnClick(_ btn: UIBarButtonItem) {
        if(btn.tag == 0){
            btn.tag = 1
            title = "demo collection view"
        }
        else{
            btn.tag = 0
            title = "demo table view"
        }
        //點擊切換功能將原本隱藏的頁面改為顯示，原本顯示的頁面改為隱藏
        tableViewContainer.isHidden = !tableViewContainer.isHidden
        collectionViewContainer.isHidden = !collectionViewContainer.isHidden
    }
}

